File: src/core/request.ts
import type { Request as UnsafeRequest } from "@literate.ink/utilities";

export class Request implements UnsafeRequest {
  public readonly url: URL;

  public constructor (path: string) {
    this.url = new URL("http://webservices-v2.crous-mobile.fr/feed/" + path);
  }
}

File: library/src/commonMain/kotlin/ink/literate/crowous/core/request.kt
package ink.literate.crowous.core

import io.ktor.client.*
import io.ktor.client.engine.cio.*
import io.ktor.client.request.*
import io.ktor.client.statement.*

class Request(path: String) {
    private val url: String = "http://webservices-v2.crous-mobile.fr/feed/$path"

    suspend fun send(): String {
        val http = HttpClient(CIO)
        val response = http.get(url)

        return response.bodyAsText()
    }
}